submissions/kt-std/friends-app/js/script.js

```diff
@@ -0,0 +1,338 @@
+	const noResultMessage = document.createElement("h3");
+	noResultMessage.innerText = "Sorry! No results found :(";
+	noResultMessage.classList.add("no-results");
+	CARDS_CONTAINER.appendChild(noResultMessage);
+}
+
+function appendFriendsCards(friendsArray) {
+	cleanCardsContainer();
+	setTotalCounter(friendsArray);
+	const fragment = document.createDocumentFragment();
+	friendsArray.forEach((friend) => {
+		const template = document.createElement("template");
+		template.innerHTML = getFriendCardTemplate(friend);
+		fragment.appendChild(template.content);
+	});
+	CARDS_CONTAINER.appendChild(fragment);
+}
+
+function appendErrorMessage(errorText) {
+	const div = document.createElement("div"),
+		img = document.createElement("img");
+	div.innerHTML = errorText;
+	div.classList.add("error__container");
+	img.classList.add("error__image");
+	img.src = "assets/error.svg";
+	div.appendChild(img);
+	document.querySelector(".main__row").style.display = "none";
+	document.querySelector(".more__button").style.display = "none";
+	document.body.append(div);
+}
+
+function cleanCardsContainer() {
+	CARDS_CONTAINER.innerHTML = "";
+}
+
+function flattenFriendProperties(friendsArray) {
+	return friendsArray.map((friend) => {
+		return {
+			firstName: friend.name.first,
+			lastName: friend.name.last,
+			email: friend.email,
+			gender: friend.gender,
+			country: friend.location.country,
+			username: friend.login.username,
+			phone: friend.phone,
+			age: friend.dob.age,
+			image: friend.picture.large,
+			registeredAge: friend.registered.age,
+			registeredDate: friend.registered.date,
+		};
+	});
+}
+
+function getFriendCardTemplate(friend) {
+	return `<div class="card__container shadow">
+				<div class="card__row around">
+					<a href='mailto:${friend.email}' class="email__button button" 
+					   data-title='${friend.email}'></a>
+					<img src="${friend.image}" class="card__image">
+					<a href='tel:${reformatPhoneNumber(friend.phone)}' class="phone__button button" 
+					   data-title='${reformatPhoneNumber(friend.phone)}'></a>
+				</div>
+				<div class="card__row column">
+					<h3 class="card__name">${friend.firstName} ${friend.lastName}</h3>
+					<h5 class="card__username">@${friend.username}</h5>
+				</div>
+				<div class="card__row gender__container">
+					<h6 class="card__gender">${getGenderIcon(friend.gender)}</h6>
+					<h6 class="card__age">${friend.age}</h6>
+				</div>
+				<div class="card__row registered__container">
+					<p class="registered__message">Friends since <br> 
+					${getDate(friend.registeredDate)}</p>
+					<div style="width:${friend.registredAge}%"></div>
+				</div>
+				<div class="card__row country__row">
+					<h6 class="card__country">${friend.country}</h6>
+				</div>
+			</div>`;
+}
+
+function reformatPhoneNumber(number) {
+	return number
+		.replace(/[^0-9]+/g, "")
+		.replace(/.(\d{3})/g, "$1-")
+		.replace(/(^\d{3,3})-(.\d+)/, "+($1)-$2")
+		.replace(/[-]+$/g, "");
```

Comment:
I will suggest you to add comments to every line which will describe what is happening

Reply:
I've added comments to explain what happens here. I would appreciate any useful tutorials about regular expressions as I'm 1000% sure that this function can be optimized. 
![beggin](https://media.giphy.com/media/vRwmQ4az7ugWk/giphy-downsized.gif)